File: real-estate-bidding-system-client/src/app/core/model/response/user/user-profile-response.model.ts
import {RoleResponseModel} from '../role/role-response.model';

export class UserProfileResponseModel {
  constructor(
    public id: number,
    public email: string,
    public firstName: string,
    public lastName: string,
    public telephone: string,
    public town: string,
    public isBanned: boolean,
    public rolesList: RoleResponseModel[]
  ) {
  }
}

File: real-estate-bidding-system-api/src/main/java/org/universe/realestatebiddingsystem/user/model/response/UserProfileResponseModel.java
package org.universe.realestatebiddingsystem.user.model.response;

import lombok.Data;

import java.util.ArrayList;
import java.util.List;

@Data
public class UserProfileResponseModel {
    private String id;
    private String email;
    private String town;
    private String telephone;
    private String firstName;
    private String lastName;
    private String isBanned;
    private List<RoleResponseModel> rolesList = new ArrayList<>();
}
